docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,854 @@
+
+[TODO: note about network effects & bridging]
+
+#### Issue 2: Flexibility / Specialization
+
+"dumb" VMs don't give you the flexibility. Any engineer knows that being able to specialize on a problem and create coarser-grained operations gives them and their users more _leverage_.  [TODO]
+
+
+Having recognized these issues, we set out to find a solution to these problems, which could allow developers to create and deploy purpose-built blockchains unified under a common source of security, with the capability of message-passing between them; a _heterogeneous sharding solution_, which we have come to know as **Parachains**.
+
+
+----
+
+## Parachains: Basic Functionality
+
+This section aims to describe, at a high level, the architecture, actors, and processes involved in the implementation of parachains. It also illuminates certain subtleties and challenges faced in the design and implementation of those processes. Our goal is to carry a parachain block from authoring to secure inclusion, and define a process which can be carried out repeatedly and in parallel for many different parachains to extend them over time. Understanding of the high-level approach taken here is important to provide context for the proposed architecture further on.
+
+The Parachain Host is a blockchain, known as the relay-chain, and the actors which provide security and inputs to the blockchain.
+
+First, it's important to go over the main actors we have involved in the parachain host.
+1. Validators. These nodes are responsible for validating proposed parachain blocks. They do so by checking a Proof-of-Validity (PoV) of the block and ensuring that the PoV remains available. They put financial capital down as "skin in the game" which can be slashed, or burned if they are proven to have misvalidated.
+2. Collators. These nodes are responsible for creating the Proofs-of-Validity that validators know how to check. Creating a PoV typically requires familiarity with the transaction format and block authoring rules of the parachain, as well as having access to the full state of the parachain.
+3. Fishermen. These are user-operated, permissionless nodes whose goal is to catch misbehaving validators in exchange for a bounty. Collators and validators can behave as Fishermen too. Fishermen aren't necessary for security, and aren't covered in-depth by this document.
+
+This alludes to a simple pipeline where collators send validators parachain blocks and their requisite PoV to check. Then, validators validate the block using the PoV, signing statements which describe either the positive or negative outcome, and with enough positive statements, the block can be included. If another validator later detects that a validator or group of validators incorrectly signed a statement claiming a block was valid, then those validators will be _slashed_, with the checker receiving a bounty.
```

Comment:
At this level of abstraction, I don't believe we have to address the semantics of what should be considered a part of the parachain or not. There are good arguments in either direction, but none should affect what the parachain host is doing.

> If a validator incorrectly signs a statement that a block was invalid, is that validator slashed?

Yes. I haven't been writing all that much AnV stuff in, but being wrong about validity or invalidity is a 100% slash.

I'll make some amendments to clarify these points

Reply:
We could permit less than 100% slash for being wrong about invalidity, but it must remain quite significant so that validator do not just vote invalidity to get others slashed, and I do not see any reason why it needs to be less than 100%.